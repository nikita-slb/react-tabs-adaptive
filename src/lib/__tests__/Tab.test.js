/* eslint getter-return: "off" */

import React from 'react';
import { Tab } from '../index';
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Tab', () => {
  it('renders properly', () => {
    const tree = renderer
    .create(
      <Tab tabName="Tab"/>
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Test click event', () => {
    const mockCallBack = jest.fn();

    const tab = shallow((<Tab tabName="Tab" onClick={mockCallBack}/>));

    tab.find('li').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('render with props', () => {
    const props = {
      tabName: 'Tab',
      tabIndex: 1,
      isActive: true,
      counter: 1
    };

    const TabComponent = mount(<Tab {...props} />);

    expect((TabComponent).prop('tabName')).toEqual(props.tabName);
    expect((TabComponent).prop('tabIndex')).toEqual(props.tabIndex);
    expect((TabComponent).prop('isActive')).toEqual(props.isActive);
    expect((TabComponent).prop('counter')).toEqual(props.counter);
  });
});
