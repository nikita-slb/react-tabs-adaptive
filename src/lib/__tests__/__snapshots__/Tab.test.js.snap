// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders properly 1`] = `
<li
  className="Tab "
  onClick={[Function]}
>
  <span
    className="Tab-name"
  >
    Tab
  </span>
</li>
`;
